src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedHandler.java

```diff
@@ -0,0 +1,239 @@
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.CLAIM_REFERENCE_NUMBER;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.DEFENDANT_NAME;
+import static uk.gov.hmcts.cmc.claimstore.services.notifications.content.NotificationTemplateParameters.FRONTEND_BASE_URL;
+import static uk.gov.hmcts.cmc.claimstore.utils.Lazy.lazily;
+
+class PaperResponseReviewedHandler {
+    private static final List<ClaimDocumentType> PAPER_RESPONSE_STAFF_UPLOADED_TYPES = List.of(
+        ClaimDocumentType.PAPER_RESPONSE_FULL_ADMIT,
+        ClaimDocumentType.PAPER_RESPONSE_DISPUTES_ALL,
+        ClaimDocumentType.PAPER_RESPONSE_PART_ADMIT,
+        ClaimDocumentType.PAPER_RESPONSE_STATES_PAID);
+
+    private static final List<String> PAPER_RESPONSE_SCANNED_TYPES = List.of("N9a", "N9b", "N11", "N225", "N180");
+
+    private static final Predicate<ClaimDocument> isPaperResponseClaimDoc = doc ->
+        PAPER_RESPONSE_STAFF_UPLOADED_TYPES.stream().anyMatch(isEqual(doc.getDocumentType()));
+
+    private static final Predicate<ScannedDocument> isPaperResponseScannedDoc = doc ->
+        PAPER_RESPONSE_SCANNED_TYPES.stream().anyMatch(type -> type.equalsIgnoreCase(doc.getSubtype()));
+
+    private static final Predicate<ClaimDocument> isClaimDocumentMoreTimeRequested = doc ->
+        ClaimDocumentType.PAPER_RESPONSE_MORE_TIME.equals(doc.getDocumentType());
+
+    private static final Predicate<ScannedDocument> isScannedDocumentMoreTimeRequested = doc ->
+        "N9".equalsIgnoreCase(doc.getSubtype());
+
+    private final CaseDetailsConverter caseDetailsConverter;
+    private final CaseMapper caseMapper;
+
+    private final ResponseDeadlineCalculator responseDeadlineCalculator;
+    private final MoreTimeRequestRule moreTimeRequestRule;
+    private final NotificationService notificationService;
+    private final NotificationsProperties notificationsProperties;
+
+    private final Lazy<Claim> claimBeforeEvent;
+    private final Lazy<Claim> claimAfterEvent;
+
+    PaperResponseReviewedHandler(
+        CaseDetailsConverter caseDetailsConverter,
+        CaseMapper caseMapper,
+        ResponseDeadlineCalculator responseDeadlineCalculator,
+        MoreTimeRequestRule moreTimeRequestRule,
+        NotificationService notificationService,
+        NotificationsProperties notificationsProperties,
+        CallbackParams callbackParams
+    ) {
+        this.caseDetailsConverter = caseDetailsConverter;
+        this.caseMapper = caseMapper;
+        this.responseDeadlineCalculator = responseDeadlineCalculator;
+        this.moreTimeRequestRule = moreTimeRequestRule;
+        this.notificationService = notificationService;
+        this.notificationsProperties = notificationsProperties;
+
+        CallbackRequest callbackRequest = callbackParams.getRequest();
+        claimBeforeEvent = lazily(() -> toClaimBeforeEvent(callbackRequest));
```

Comment:
What is the rationale for using `lazily`?

Reply:
To stop it being calculated multiple times when it will give the same result